src/solvers/dg_unstructured_quad/dg_2d.jl

```diff
@@ -0,0 +1,341 @@
+function calc_boundary_flux!(cache, t, boundary_condition, equations, dg::DG, initial_condition, mesh::UnstructuredQuadMesh)
+
+  @unpack surface_flux = dg
+  @unpack normals, tangents, scaling, surface_flux_values = cache.elements
+  @unpack u, element_id, element_side_id, node_coordinates, name  = cache.boundaries
+
+  @threaded for boundary in eachboundary(cache.boundaries)
+    # Get the element and side IDs on the primary element
+    primary_element = element_id[boundary]
+    primary_side    = element_side_id[boundary]
+
+    for i in eachnode(dg)
+      # hacky way to set "exact solution" boundary conditions. Only used to test the orientation
+      # for a mesh with flipped elements
+      u_ext = initial_condition((node_coordinates[1, i, boundary],
+                                 node_coordinates[2, i, boundary]),
+                                 t, equations)
+
+      # pull the left state from the boundary u values on the primary element as well as the
+      # directional vectors and scaling
+      #   Note! this assumes a conforming approximation, more must be done in terms of the normals
+      #         and tangents for hanging nodes and other non-conforming approximation spaces
+      u_ll  = u[1, :, i, boundary]
+      n_vec = normals[ :, i, primary_side, primary_element]
+      t_vec = tangents[:, i, primary_side, primary_element]
+      scal  = scaling[i, primary_side, primary_element]
+
+      # rotate states
+      u_tilde_ll  = rotate_solution(u_ll , n_vec, t_vec, equations)
+      u_tilde_ext = rotate_solution(u_ext, n_vec, t_vec, equations)
+
+      # Call pointwise Riemann solver in the rotated direction
+      flux_tilde = surface_flux(u_tilde_ll, u_tilde_ext, 1, equations)
+
+      # backrotate the flux into the original direction
+      flux = backrotate_flux(flux_tilde, n_vec, t_vec, equations)
+
+      # Scale the flux appropriately and copy back to primary element storage
+      for v in eachvariable(equations)
+        surface_flux_values[v, i, primary_side, primary_element] = flux[v] * scal
+      end
+    end
+  end
+
+  return nothing
+end
+
+# Note! The local side numbering for the unstructured quadrilateral element implementation differs
+#       from the strcutured TreeMesh or CurveMesh local side numbering:
+#
+#      TreeMesh/CurveMesh sides   versus   UnstructuredMesh sides
+#                  4                                  3
+#          -----------------                  -----------------
+#          |               |                  |               |
+#          | ^ eta         |                  | ^ eta         |
+#        1 | |             | 2              4 | |             | 2
+#          | |             |                  | |             |
+#          | ---> xi       |                  | ---> xi       |
+#          -----------------                  -----------------
+#                  3                                  1
```

Comment:
Thanks a lot for this comment! This should probably also go into the docs...

Reply:
Yeah, I figured this comment would probably move to a more appropriate place once we start to iterate on this implementation

Reply:
I wouldn't necessarily move it... just leave the comment here and maybe add a note that there's more description in the docs?